docs/reference/pip_install.rst

```diff
@@ -36,6 +39,33 @@ which depends on foo::
     ...
     Installing collected packages foo, baz, bar
 
+
+Prior to v6.1.0, pip made no commitments about install order.
+
+The decision to install topologically is based on the principle that
+installations should proceed in a way that leaves the environment usable at each
+step. This has two main practical benefits:
+
+1. Concurrent use of the environment during the install is more likely to work.
+2. A failed install won't leave broken packages.  Although pip would like to
```

Comment:
just to be clear, of the 6 you listed, only 1 it seems (the fragmented permissions case) is something that would actually involve an installation error which is the sense of "failed install" I was thinking of.

also, I don't see how the "missing things needed to build" would result in a broken package... wouldn't it just fail to complete the build portion of the install, and never get to copying files into the target environment.

regardless, I'm still going with the "less likely" language.